libraries/ms-common/src/stm32f0xx/soft_timer.c

```diff
@@ -0,0 +1,95 @@
+#include "soft_timer.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+
+#include "status.h"
+#include "stm32f0xx_rcc.h"
+#include "stm32f0xx_tim.h"
+
+typedef struct timer {
+  uint32_t duration;
+  uint32_t time_started;
+  soft_timer_callback callback;
+  bool inuse;
+  bool rolled_over;
+  void *context;
+} timer;
+
+static timer s_soft_timer_array[SOFT_TIMER_NUM];
```

Comment:
I'm not a fan of having so many static variables internal to the module. I'm pretty sure your tests would fail if you added a test after your exhaustive one.